src/app-layout/visual-refresh-toolbar/toolbar/trigger-button/index.tsx

```diff
@@ -44,11 +89,103 @@ function TriggerButton(
     badge,
     selected = false,
     highContrastHeader,
+    hasTooltip = false,
+    hideTooltipOnFocus = false,
+    tooltipText,
+    isForPreviousActiveDrawer = false,
+    hasOpenDrawer = false,
+    isMobile = false,
   }: TriggerButtonProps,
   ref: React.Ref<ButtonProps.Ref>
 ) {
+  const containerRef = useRef(null);
+  const tooltipValue = tooltipText ? tooltipText : ariaLabel ? ariaLabel : '';
+  const [showTooltip, setShowTooltip] = useState<boolean>(false);
+  //there is an override to used to hide the tooltip on certan programatic focus events
+  const [showTooltipOverride, setShowTooltipOverride] = useState<boolean>(false);
+
+  const onShowTooltipSoft = (show: boolean) => {
+    setShowTooltipOverride(false);
+    setShowTooltip(show);
+  };
+
+  const onShowTooltipHard = (show: boolean) => {
+    if (!showTooltipOverride) {
+      setShowTooltip(show);
+    }
+  };
+
+  const handleOnClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    console.log('handle on click');
```

Comment:
To be removed

Reply:
i will add an eslint rule for these in a separate PR